File: app/src/debug/java/com/sakuna63/tumblrin/application/di/component/DebugApplicationComponent.java
package com.sakuna63.tumblrin.application.di.component;

import com.sakuna63.tumblrin.DebugApp;
import com.sakuna63.tumblrin.application.di.module.ApplicationModule;
import com.sakuna63.tumblrin.application.di.module.AuthenticationModule;
import com.sakuna63.tumblrin.application.di.module.DebugApiModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {ApplicationModule.class, AuthenticationModule.class, DebugApiModule.class})
public interface DebugApplicationComponent extends ApplicationComponent {
    void inject(DebugApp app);
}

File: app/src/main/java/com/sakuna63/tumbin/application/di/component/DebugApplicationComponent.kt
package com.sakuna63.tumbin.application.di.component

import com.sakuna63.tumbin.DebugApp
import com.sakuna63.tumbin.application.di.module.ApiModule
import com.sakuna63.tumbin.application.di.module.ApplicationModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class, ApiModule::class))
interface DebugApplicationComponent : ApplicationComponent {
    fun inject(app: DebugApp)
}
